### PR TITLE
Issue/501 metadata to batches

### LINKED ISSUE #501: add metadata to batches
## Detailed Description
Add 
- t0
- x_centers
- y_centers
to the batches

## Context
- Critical that predictions save the t0_dt
- Useful for ML models to know these values

## Possible Implementation
when loading the batches, load the row from the .csv file and store in pydantic object

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Add metadata to batch
The metadata is loaded from the csv file and contains
- t0_dts
- x centers
- y centers
- the batch size of the batch, i,.e how many examples there are

Fixes #501 

## How Has This Been Tested?

- added unittests
- normal unittest

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -71,9 +70,9 @@ def hrv_sat_data_source(hrv_sat_filename: Path):  # noqa: D103
     )
 
 
-@pytest.fixture
-def general_data_source():  # noqa: D103
-    return MetadataDataSource(history_minutes=0, forecast_minutes=5, object_at_center="GSP")
+# @pytest.fixture
+# def general_data_source():  # noqa: D103
```
Comment (jacobbieker) [inline conftest.py:74]: Could we delete this?

**Review by jacobbieker — APPROVED** [review]
LGTM!

File: nowcasting_dataset/data_sources/metadata/metadata_model.py [hunk 2]
```diff
@@ -1,11 +1,96 @@
...
+        """
+        Save metadata to a csv file
+
+        Args:
+            path: the path where the file shold be save
+
+        """
+
+        filename = f"{path}/{SPATIAL_AND_TEMPORAL_LOCATIONS_OF_EACH_EXAMPLE_FILENAME}"
+        metadata_dict = self.dict()
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/metadata/metadata_model.py:49]: will this result in every row in the CSV containing the batch size?

File: nowcasting_dataset/data_sources/metadata/metadata_model.py [hunk 3]
```diff
@@ -1,11 +1,96 @@
...
+        path: the path which stores the metadata file
+        batch_idx: the batch index
+        batch_size: how many examples in each batch
+
+    Returns: Metadata class
+    """
+    filename = f"{path}/{SPATIAL_AND_TEMPORAL_LOCATIONS_OF_EACH_EXAMPLE_FILENAME}"
+
+    # read whole file
+    metadata_df = pd.read_csv(filename)
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/metadata/metadata_model.py:82]: if it's slow to read the entire CSV then consider using something like `read_csv(skiprows=start_example_idx+1, nrows=batch_size)`  (although I'm not sure if `skiprows` also skips the CSV header!)  But don't bother with this if reading the whole CSV file is fast enough!

**Review by JackKelly — APPROVED** [review]
LGTM!

File: conftest.py (hunk 1 above)
Comment (peterdudfield) [reply conftest.py]: thanks

File: nowcasting_dataset/data_sources/metadata/metadata_model.py (hunk 2 above)
Comment (peterdudfield) [reply nowcasting_dataset/data_sources/metadata/metadata_model.py:49]: metadata_dict.pop("batch_size")

the next line down gets rid of batch size

File: nowcasting_dataset/data_sources/metadata/metadata_model.py (hunk 3 above)
Comment (peterdudfield) [reply nowcasting_dataset/data_sources/metadata/metadata_model.py:82]: That seems worth doing

File: nowcasting_dataset/data_sources/metadata/metadata_model.py (hunk 2 above)
Comment (JackKelly) [reply nowcasting_dataset/data_sources/metadata/metadata_model.py:49]: oops, right you are, sorry! I clearly haven't had enough coffee yet this morning!

**Pull request merged**